cmd/collectors/zapi/collector/zapi.go

```diff
@@ -251,15 +252,22 @@ func (z *Zapi) PollData() (map[string]*matrix.Matrix, error) {
 		oldInstances.Add(key)
 	}
 
-	fetch = func(instance *matrix.Instance, node *node.Node, path []string) {
+	fetch = func(instance *matrix.Instance, node *node.Node, path []string, isAppend bool) {
 
 		newpath := append(path, node.GetNameS())
 		key := strings.Join(newpath, ".")
 		z.Logger.Trace().Msgf(" > %s(%s)%s <%s%d%s> name=[%s%s%s%s] value=[%s%s%s]", color.Grey, newpath, color.End, color.Red, len(node.GetChildren()), color.End, color.Bold, color.Cyan, node.GetNameS(), color.End, color.Yellow, node.GetContentS(), color.End)
+
 		if value := node.GetContentS(); value != "" {
 			if label, has := z.instanceLabelPaths[key]; has {
-				instance.SetLabel(label, value)
-				z.Logger.Trace().Msgf(" > %slabel (%s) [%s] set value (%s)%s", color.Yellow, key, label, value, color.End)
+				// Handling array with comma separated values
+				if isAppend && instance.GetLabel(label) != "" {
```

Comment:
might be better to cache `instance.GetLabel(label)` and only call once

Reply:
done